trieste/acquisition/optimizer.py

```diff
@@ -35,24 +34,91 @@
 AcquisitionOptimizer = Callable[[SP, AcquisitionFunction], TensorType]
 """
 Type alias for a function that returns the single point that maximizes an acquisition function over
-a search space. For a search space with points of shape S, and acquisition function with input shape
-[...] + S output shape [..., 1], the :data:`AcquisitionOptimizer` return shape should be [1] + S.
+a search space. For a search space with points of shape [D], and acquisition function with input
```

Comment:
either [..., D] and [...., B, D] or [D] and [B, D]?

Reply:
actually maybe it's [..., 1, D] now?

Reply:
it's always [..., B, D] but in some cases B = 1